sklearn/inspect/tests/test_partial_dependence.py

```diff
@@ -0,0 +1,533 @@
+@pytest.mark.parametrize('power', (1, 2))
+def test_partial_dependence_easy_target(est, power):
+    # If the target y only depends on one feature in an obvious way (linear or
+    # quadratic) then the partial dependence for that feature should reflect
+    # it.
+    # We here fit a linear regression_data model (with polynomial features if
+    # needed) and compute r_squared to check that the partial dependence
+    # correctly reflects the target.
+
+    rng = np.random.RandomState(0)
+    n_samples = 100
+    target_variable = 2
+    X = rng.normal(size=(n_samples, 5))
+    y = X[:, target_variable]**power
+
+    est.fit(X, y)
+
+    averaged_predictions, values = partial_dependence(
+        est, features=[target_variable], X=X, grid_resolution=1000)
+
+    new_X = values[0].reshape(-1, 1)
+    new_y = averaged_predictions[0]
+    # add polynomial features if needed
+    new_X = PolynomialFeatures(degree=power).fit_transform(new_X)
+
+    lr = LinearRegression().fit(new_X, new_y)
+    r2 = r2_score(new_y, lr.predict(new_X))
+
+    assert r2 > .99
+
+
+@pytest.mark.filterwarnings('ignore:The default value of ')  # 0.22
+@pytest.mark.parametrize('Estimator',
+                         (sklearn.tree.DecisionTreeClassifier,
+                          sklearn.tree.ExtraTreeClassifier,
+                          sklearn.ensemble.ExtraTreesClassifier,
+                          sklearn.neighbors.KNeighborsClassifier,
+                          sklearn.neighbors.RadiusNeighborsClassifier,
+                          sklearn.ensemble.RandomForestClassifier))
+def test_multiclass_multioutput(Estimator):
+    # Make sure error is raised for multiclass-multioutput classifiers
+
+    # make multiclass-multioutput dataset
+    X, y = make_classification(n_classes=3, n_clusters_per_class=1,
+                               random_state=0)
+    y = np.array([y, y]).T
+
+    est = Estimator()
+    est.fit(X, y)
+
+    assert_raises_regex(ValueError,
+                        "Multiclass-multioutput estimators are not supported",
+                        partial_dependence, est, [0], X=X)
+
+
+def test_partial_dependence_input():
+    # Test input validation of partial_dependence.
+
+    X, y = make_classification(random_state=0)
+
+    lr = LinearRegression()
+    lr.fit(X, y)
+    gbc = GradientBoostingClassifier(random_state=0)
+    gbc.fit(X, y)
+
+    assert_raises_regex(ValueError,
+                        "est must be a fitted regressor or classifier",
+                        partial_dependence, KMeans(), [0], X)
+
+    assert_warns_message(
+        UserWarning,
+        'The response_method parameter is ignored for regressors',
+        partial_dependence, lr, [0], X, response_method='predict_proba')
+
+    assert_raises_regex(
+        ValueError,
+        "With the 'recursion' method, the response_method must be "
+        "'decision_function'.",
+        partial_dependence, gbc, [0], X, response_method='predict_proba',
+        method='recursion')
```

Comment:
We also need to test the `estimator=gbc, response_method='predict_proba', method='auto'` case. I don't care if it works by brute or if it fails complaining that recursion and predict_proba aren't friends, but it should be tested.

Reply:
Indeed, thanks.
I chose to error because it makes the logic simpler